lib/client.js

```diff
@@ -362,11 +362,28 @@ class Client extends EventEmitter {
         USERAUTH_FAILURE: (p, authMethods, partialSuccess) => {
           if (curAuth.type === 'agent') {
             const pos = curAuth.agentCtx.pos();
-            debug && debug(`Client: Agent key #${pos + 1} failed`);
-            return tryNextAgentKey();
+            if (partialSuccess) {
+              const key = curAuth.agentCtx.currentKey();
+              proto.authPK(curAuth.username, key, (buf, cb) => {
+              curAuth.agentCtx.sign(key, buf, {}, (err, signed) => {
+                if (err) {
+                  err.level = 'agent';
+                  this.emit('error', err);
+                } else {
+                  return cb(signed);
+                }
+
+                return tryNextAgentKey();
+              });
+            });
```

Comment:
@lucasvbeek is this actually needed? I'm cherry picking this PR and noticed that with this `USERAUTH_FAILURE` gets called again, just checking for `partialSuccess` seems enough and I can connect successfully.
```
if (!partialSuccess) {
      debug && debug(`Client: Agent key #${pos + 1} failed`);
      return tryNextAgentKey();
}
```

Reply:
Looks like it should be fine with just the check, I copied the full block from `USERAUTH_PK_OK` to make sure I didn't miss anything. Thanks for checking